### ISSUE TITLE
models pretrained on DFDC works well, but why models pertrained on FFPP always give a low score?

### ISSUE THREAD

**Issue by 2258324319** [polimi-ispl/icpr2020dfdc#74]
I tried to feed true videos and fake videos to the models pre-trained on the FFPP, but it always gave scores less than 0.2. Is that normal? :(

**Comment by CrohnEngineer** [comment]
Hi @2258324319 ,

Sorry for the very late reply 🙏 
The phenomenon you encountered is very complex but can be summarized in two words: performance generalization, i.e., it is not guaranteed that deep learning solutions trained on some data can generalize their performances on other datasets.  
As far as I know, the DFDC is a "messier" dataset, with lots of noise in both the single videos and labels, while FF++ represents a more "controlled" scenario.  
Therefore, this might explain why the models trained on DFDC performed better on your videos.  

However, remember that neural networks are sensitive to the type of data they are working with, and this is even more true for forensic tasks like deepfake detection. 
The features neural networks look for in these cases are subtle and lie on a more signal processing level rather than a high human comprehensible level. To make it simple, it is easier for a neural network trained to recognize dogs and cats to generalize over images from different datasets rather than recognize deepfake videos generated with different techniques :)

If you are interested in the topic, we wrote a [paper](https://ieeexplore.ieee.org/abstract/document/9360901) some time ago about the best techniques for making a CNN generalize over different datasets.

I hope this answers your question! Feel free to ask for any other information.
